test/pylib/cpp/boost/prepare_instance.py

```diff
@@ -0,0 +1,93 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+from __future__ import annotations
+
+import asyncio
+import logging
+import os
+import time
+from pathlib import Path
+
+from test.pylib.host_registry import HostRegistry
+from test.pylib.minio_server import MinioServer
+from test.pylib.s3_server_mock import MockS3Server
+from test.pylib.util import LogPrefixAdapter
+
+
+class PrepareWorker:
+    def __init__(self, env_file: Path):
+        self.env_file = env_file
+
+    def prepare(self) -> None:
+        timeout = 10
+        start_time = time.time()
+        while True:
+            if os.path.exists(self.env_file):
+                with open(self.env_file, 'r') as file:
+                    for line in file.readlines():
+                        key, value = line.strip().split('=', 1)
+                        os.environ[key] = value
+                break
+
+            if time.time() - start_time > timeout:
+                raise TimeoutError(f"Timeout waiting for file {self.env_file}")
+            # Sleep needed to wait when the controller will create a file with environment variables.
+            # Without sleep checking of the file existence will be too fast,
+            # so it will finish before the file is created
+            time.sleep(0.5)
+
+    def cleanup(self) -> None:
+        pass
+
+
+class PrepareController:
```

Comment:
Please add a class comment. What does this class do? Ithink the name could be more telling.